internal/resources/grafana/resource_data_source_test.go

```diff
@@ -277,12 +312,22 @@ func testAccDataSourceCheckExists(rn string, dataSource *gapi.DataSource) resour
 			return fmt.Errorf("resource id not set")
 		}
 
-		id, err := strconv.ParseInt(rs.Primary.ID, 10, 64)
+		orgID, idStr := grafana.SplitOrgResourceID(rs.Primary.ID)
+		id, err := strconv.ParseInt(idStr, 10, 64)
 		if err != nil {
 			return fmt.Errorf("resource id is malformed")
 		}
 
 		client := testutils.Provider.Meta().(*common.Client).GrafanaAPI
+		// If the org ID is set, check that the datasource doesn't exist in the default org
+		if orgID > 1 {
+			datasource, err := client.DataSource(id)
+			if err == nil || datasource != nil {
```

Comment:
Could it ever happen that both `err` and `datasource` are nil?

Reply:
I don't think so but it'll throw an error so we'll definitely see something is wrong